extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -35,6 +35,46 @@ const BUILT_IN_FUNCTIONS = 'built-in-functions';
  * @const @private {!Object<string, !Object<string, Function>>}
  */
 const FUNCTION_WHITELIST = (function() {
+
+  /**
+   * Custom wrappers for mutating array methods so that we can offer the
+   * same functionality without allowing mutation on variables in bind scope.
+   */
+  const BindArrays = {
+    /**
+     * Similar to Array.prototype.splice, except it returns a copy of the
+     * passed-in array with the desired modifications.
+     * @param {!Array} array
+     * @param {number=} start
+     * @param {number=} end
+     * @param {...?} items
+     */
+    /*eslint "no-unused-vars": 0*/
+    'arraySplice': function(array, start, end, items) {
+      const copy = array.slice();
+      copy.splice.apply(copy, Array.prototype.slice.call(arguments, 1));
+      return copy;
+    },
+
+    /**
+     * Similar to array[index] = item except it returns a copy of the
+     * passed-in array with the desired modification.
+     * @param {!Array} array
+     * @param {!number} index, must be an integer
+     * @param {?=} item, null if unspecified
+     */
+    'arraySet': function(array, index, item) {
```

Comment:
I was thinking that too. I tried it out and we would need quite a bit of special logic to make the call to splice to act as expected. The code as-is is shorter and easier to read IMO.